user/contactmethod/store.go

```diff
@@ -124,6 +128,13 @@ func NewDB(ctx context.Context, db *sql.DB) (*DB, error) {
 				DELETE FROM user_contact_methods
 				WHERE id = any($1)
 			`),
+		findLastStatus: p.P(`
+			SELECT last_status, status_details, next_retry_at notnull
+			FROM outgoing_messages
+			WHERE contact_method_id = $1 AND message_type = 'test_notification'
+			ORDER BY created_at DESC
+			LIMIT 1
```

Comment:
I wonder if we should store the message ID in the contact method table as part of the send operation rather than needing to search

Has there been any perf. testing/analyze done with stage data? I don't think there is any type of index on `created_at` and with this being a mostly unbounded API -- that could cause big problems.

Let's have a chat on our options